distribution/lib/Standard/Database/0.0.0-dev/src/DB_Table.enso

```diff
@@ -3287,6 +3295,25 @@ make_literal_table connection column_vectors column_names alias =
 
             DB_Table.new alias connection internal_columns context
 
+## PRIVATE
+private _resolve_grouping_columns columns_helper:Table_Column_Helper group_by:Vector problem_builder:Problem_Builder -> Vector =
+    grouping_columns = columns_helper.select_columns_helper group_by Case_Sensitivity.Default True problem_builder
+    grouping_columns.each column->
+        if column.value_type.is_floating_point then
+            problem_builder.report_other_warning (Floating_Point_Equality.Error column.name)
+    grouping_columns
+
+## PRIVATE
+private _resolve_order_descriptors dialect ordering default_ordering =
+    case ordering.is_empty of
+        False -> ordering.map element->
+            column = element.column
+            associated_selector = element.associated_selector
+            dialect.prepare_order_descriptor column associated_selector.direction text_ordering=Nothing
+        True -> case default_ordering of
+            Nothing -> Error.throw (Illegal_Argument.Error "The table has no existing ordering (e.g. from a `sort` operation or primary key). `offset` requires an ordering in database.")
```

Comment:
This is an `offset`-centric error message, but isn't this method also used from `add_row_number`?

Reply:
Ah yes... I lost that as I refactored away the code I had previously duplicated :) I will add it back.

Reply:
done